iceoryx_dds/test/moduletests/test_fastdds.cpp

```diff
@@ -0,0 +1,193 @@
+using TestDataReader = FastDDSDataReader;
+using TestDataWriter = FastDDSDataWriter;
+
+// ======================================== Fixture ======================================== //
+class FastDDSTest : public Test
+{
+  public:
+    void SetUp(){};
+    void TearDown(){};
+};
+
+// ======================================== Tests ======================================== //
+TEST_F(FastDDSTest, DoesNotAttemptToReadWhenDisconnected)
+{
+    // ===== Setup
+    uint64_t bufferSize = 1024;
+    uint8_t buffer[bufferSize];
+
+    // ===== Test
+    TestDataReader reader{"", "", ""};
+
+    auto takeResult = reader.take(buffer, bufferSize, iox::cxx::nullopt);
+    EXPECT_EQ(true, takeResult.has_error());
+    EXPECT_EQ(iox::dds::DataReaderError::NOT_CONNECTED, takeResult.get_error());
+
+    auto takeNextResult = reader.takeNext(buffer, bufferSize);
+    EXPECT_EQ(true, takeNextResult.has_error());
+    EXPECT_EQ(iox::dds::DataReaderError::NOT_CONNECTED, takeResult.get_error());
+}
+
+TEST_F(FastDDSTest, ReturnsErrorWhenAttemptingToReadIntoANullBuffer)
+{
+    // ===== Setup
+    uint64_t bufferSize = 0;
+    uint8_t* buffer = nullptr;
+
+    // ===== Test
+    TestDataReader reader{"", "", ""};
+    reader.connect();
+
+    auto takeResult = reader.take(buffer, bufferSize, iox::cxx::nullopt);
+    EXPECT_EQ(true, takeResult.has_error());
+    EXPECT_EQ(iox::dds::DataReaderError::INVALID_RECV_BUFFER, takeResult.get_error());
+
+    auto takeNextResult = reader.takeNext(buffer, bufferSize);
+    EXPECT_EQ(true, takeNextResult.has_error());
+    EXPECT_EQ(iox::dds::DataReaderError::INVALID_RECV_BUFFER, takeNextResult.get_error());
+}
+
+TEST_F(FastDDSTest, SendAndTakeNext)
+{
+    uint64_t bufferSize = 1024;
+
+    /* Create reader */
+    TestDataReader reader{"test", "test", "test"};
+    reader.connect();
+
+    /* Create writer */
+    TestDataWriter writer{"test", "test", "test"};
+    writer.connect();
+
+    /* Wait for discovery */
+    writer.waitForReaderDiscovery(1);
+    reader.waitForWriterDiscovery(1);
+
+    /* Setup send buffer */
+    uint8_t sendBuffer[bufferSize];
+    // Populate send buffer with random values
+    std::srand(static_cast<uint>(std::time(nullptr)));
+    for (uint64_t i = 0; i < bufferSize; i++)
+    {
+        sendBuffer[i] = static_cast<uint8_t>(std::rand() % 255);
+    }
+
+    /* Send data */
+    writer.write(sendBuffer, bufferSize);
+
+    /* Wait for delivery */
+    reader.waitForData();
+
+    /* Setup receive buffer */
+    uint8_t receiveBuffer[bufferSize];
+    for (uint64_t i = 0; i < bufferSize; i++)
+    {
+        receiveBuffer[i] = 0;
+    }
+
+    /* Take data */
+    auto result = reader.takeNext(receiveBuffer, bufferSize);
+
+    /* Check correctness */
+    EXPECT_EQ(false, result.has_error());
+    for (uint64_t i = 0; i < bufferSize; i++)
+    {
+        ASSERT_EQ(sendBuffer[i], receiveBuffer[i]);
+    }
+}
+
+TEST_F(FastDDSTest, SendAndTake)
```

Comment:
I think we also need tests for calling `take` or `takeNext` when no more samples are available.
Is it also possible to test the case when `iox::dds::DataReaderError::INVALID_DATA` was received?